backend/internal/db/dbgorm/triggers.go

```diff
@@ -0,0 +1,105 @@
+package dbgorm
+
+import (
+	"context"
+	"fmt"
+
+	"gorm.io/gorm/clause"
+
+	"go.autokitteh.dev/autokitteh/backend/internal/db/dbgorm/scheme"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+func (db *gormdb) triggerToRecord(ctx context.Context, trigger sdktypes.Trigger) (*scheme.Trigger, error) {
+	connID := sdktypes.GetTriggerConnectionID(trigger)
+
+	conn, err := db.GetConnection(ctx, connID)
+	if err != nil {
+		return nil, fmt.Errorf("get trigger connection: %w", err)
+	}
+
+	projID := sdktypes.GetConnectionProjectID(conn)
+
+	envID := sdktypes.GetTriggerEnvID(trigger)
+	if envID != nil {
+		env, err := db.GetEnvByID(ctx, envID)
+		if err != nil {
+			return nil, fmt.Errorf("get trigger env: %w", err)
+		}
+
+		if projID.String() != sdktypes.GetEnvProjectID(env).String() {
+			return nil, fmt.Errorf("env and connection project mismatch: %s != %s", projID.String(), sdktypes.GetEnvProjectID(env).String())
+		}
+	}
+
+	return &scheme.Trigger{
+		TriggerID:    sdktypes.GetTriggerID(trigger).String(),
+		EnvID:        envID.String(),
+		ProjectID:    projID.String(),
+		ConnectionID: connID.String(),
+		EventType:    sdktypes.GetTriggerEventType(trigger),
+		CodeLocation: sdktypes.GetCodeLocationCanonicalString(sdktypes.GetTriggerCodeLocation(trigger)),
+	}, nil
+}
+
+func (db *gormdb) CreateTrigger(ctx context.Context, trigger sdktypes.Trigger) error {
+	t, err := db.triggerToRecord(ctx, trigger)
+	if err != nil {
+		return err
+	}
+
+	// Assuming if eventID already exists, nothing will happen and no error
+	if err := db.db.WithContext(ctx).Clauses(clause.OnConflict{DoNothing: true}).Create(t).Error; err != nil {
```

Comment:
```suggestion
	if err := db.db.WithContext(ctx).Create(t).Error; err != nil {
```

we want to fail if you try to create a trigger with the same id again

updates should work

Reply:
@efiShtain not sure I agree. This is valid idempotent behavior, just like no-op updates.

Reply:
i think i added it back then when i thought we might have to retry adding with the same id. I'm going to adopt efi's proposal here and remove it - this shouldn't happen.